src/main/ruby/truffleruby/core/array.rb

```diff
@@ -296,11 +296,45 @@ def dig(idx, *more)
     if Primitive.nil?(result) || more.empty?
       result
     else
-      raise TypeError, "#{result.class} does not have #dig method" unless result.respond_to?(:dig)
-      result.dig(*more)
+      Array.looping_dig(result, *more)
     end
   end
 
+  def self.looping_dig(obj, *idxs)
+    idxs.each_with_index do |idx, n|
+      return obj if Primitive.nil?(obj)
+      case obj
+      when Hash
+        if basic_method?(obj, :dig)
```

Comment:
We should avoid a call here, either inline the helper or even better a new Primitive doing both `Primitive.vm_method_is_basic(someMethod)` + the lookup.